Code/Framework/AzCore/AzCore/Math/RayIntersection.h

```diff
@@ -0,0 +1,21 @@
+
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include "AzCore/Math/Vector2.h"
+namespace AZ
+{
+    class Ray;
+    class Vector3;
+
+    namespace RayIntersection
```

Comment:
update: I would strongly recommend against putting this in a nested namespace. This is much easier to use just being in `AZ`, annoyingly as we do use a nested namespace already we should use `AZ::Intersect` here to be consistent with `IntersectSegment.h`